File: RQ3/Java-REval-Desc/task70_desc.java
public class FileNameCheck {
    public static void file_name_check(String file_name) {
        String[] suf = {"txt", "exe", "dll"};
        String[] lst = file_name.split("\\.");
        String _result = "Yes";
        if (lst.length != 2) {
            _result = "No";
        }
        boolean isValidSuffix = false;
        for (String suffix : suf) {
            if (lst[1].equals(suffix)) {
                isValidSuffix = true;
                break;
            }
        }
        if (!isValidSuffix) {
            _result = "No";
        }
        if (lst[0].length() == 0) {
            _result = "No";
        }
        if (!Character.isAlphabetic(lst[0].charAt(0))) {
            _result = "No";
        }
        long t = lst[0].chars().filter(Character::isDigit).count();
        if (t > 3) {
            _result = "No";
        }

        assert func(_result);  // POST: func is equivalent to: Create a function which takes a string representing a file's name, and returns 'Yes' if the file's name is valid, and returns 'No' otherwise. A file's name is considered to be valid if and only if all the following conditions are met: - There should not be more than three digits ('0'-'9') in the file's name. - The file's name contains exactly one dot '.' - The substring before the dot should not be empty, and it starts with a letter the latin alphapet ('a'-'z' and 'A'-'Z'). - The substring after the dot should be one of these: ['txt', 'exe', 'dll'] Examples: file_name_check("example.txt") # => 'Yes' file_name_check("1example.dll") # => 'No' (the name should start with a latin alphapet letter)
    }
}

File: RQ3/C-REval-Desc/task70_desc.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int func(const char *result);

void file_name_check(const char *file_name) {
    const char *suf[] = {"txt", "exe", "dll"};
    char *lst[2];
    char *file_name_copy = strdup(file_name);
    char *_result = "Yes";
    
    lst[0] = strtok(file_name_copy, ".");
    lst[1] = strtok(NULL, ".");

    if (lst[1] == NULL || lst[0] == NULL || strtok(NULL, ".") != NULL) {
        _result = "No";
    }
    if (lst[1] != NULL && (strcmp(lst[1], suf[0]) != 0 && strcmp(lst[1], suf[1]) != 0 && strcmp(lst[1], suf[2]) != 0)) {
        _result = "No";
    }
    if (strlen(lst[0]) == 0) {
        _result = "No";
    }
    if (!isalpha(lst[0][0])) {
        _result = "No";
    }
    
    int digit_count = 0;
    for (int i = 0; i < strlen(lst[0]); i++) {
        if (isdigit(lst[0][i])) {
            digit_count++;
        }
    }
    if (digit_count > 3) {
        _result = "No";
    }

    assert(func(_result));
    free(file_name_copy);
}